openverse-api/catalog/api/utils/waveform.py

```diff
@@ -0,0 +1,131 @@
+import json
+import logging
+import math
+import mimetypes
+import os
+import pathlib
+import shutil
+import subprocess
+
+import requests
+
+log = logging.getLogger(__name__)
+
+TMP_DIR = pathlib.Path('/tmp').resolve()
+UA_STRING = 'OpenverseWaveform/0.0 (https://wordpress.org/openverse)'
+
+
+def ext_from_url(url):
+    """
+    Get the file extension from the given URL. Looks at the last part of the URL
+    path, and returns the string after the last dot.
+
+    :param url: the URL to the file whose extension is being determined
+    :returns: the file extension or ``None``
+    """
+    file_name = url.split('/')[-1]
```

Comment:
Can a url have no '/' in it? Probably not, since it's a full url from the catalog. But if yes, this will raise an error.

Reply:
Since the URLs point to the actual file which are very unlikely to be at the domain root, it'll most likely have the `/` in it.